http-server/src/main/java/io/micronaut/http/server/OptionsFilter.java

```diff
@@ -68,13 +67,15 @@ public HttpResponse<?> filterRequest(HttpRequest<?> request) {
         if (hasOptionsRouteMatch(request)) {
             return null; // proceed
         }
-        MutableHttpResponse<?> mutableHttpResponse = HttpResponse.status(HttpStatus.OK);
-        router.findAny(request.getUri().toString(), request)
-            .map(UriRouteMatch::getHttpMethod)
-            .map(HttpMethod::toString)
-            .forEach(allow -> mutableHttpResponse.header(HttpHeaders.ALLOW, allow));
-        mutableHttpResponse.header(HttpHeaders.ALLOW, HttpMethod.OPTIONS.toString());
-        return mutableHttpResponse;
+        if (HttpStatus.METHOD_NOT_ALLOWED.equals(response.getStatus())) {
+            List<String> allowedMethods = response.getHeaders().get(HttpHeaders.ALLOW, String[].class)
```

Comment:
don't we have a getAll method to simplify this

Reply:
IMO, `getAll` makes it more complicated because it does not do any parsing of the values, it just returns the raw value strings. So in this case, the header has already been written in `RequestLifecycle` as something like `Allow: GET, POST, HEAD`. `getAll` then returns a list with a single `String` value of `"GET,POST,HEAD"`. Thus to produce this same result we would need conditional logic to check whether that `String` is empty to determine whether to add a comma before appending `OPTIONS` to it.